homeassistant/helpers/event.py

```diff
@@ -565,5 +577,5 @@ def process_state_match(
     if isinstance(parameter, str) or not hasattr(parameter, "__iter__"):
         return lambda state: state == parameter
 
-    parameter_tuple = tuple(parameter)
-    return lambda state: state in parameter_tuple
+    parameter_set = set(parameter)
```

Comment:
Super teeny tiny extra optimization:

```suggestion
    parameter_set = set(parameter)
    
    if len(parameter_set) == 1:
        value = next(iter(parameter_set))
        return lambda state: state == value
```

I guess we can also return `value.__eq__` but I don't know if that matters.

Reply:
I left this one as-is per discussion on Discord